### DISCUSSION TITLE
.Net nuget package?

### DISCUSSION THREAD

**Discussion by rurounijones** [coqui-ai/STT#2027] [Q&A] [answered]
I am probably missing something here but is there a nuget package for Coqui STT?

There are the old Mozilla deepspeech packages available in the nuget package manager for Visual Studio:

![image](https://user-images.githubusercontent.com/21911/142971244-7ff3e9f5-dece-427a-bd92-4558cfc56bdc.png)

But I cannot find any hint of coqui STT packages and couldn't find anything obvious on the [readthedocs](https://stt.readthedocs.io/en/latest/DotNet-Examples.html) website (there is an API reference and usage example but that assumes you already have it installed. Nothing about actually adding it). and there is nothing in https://github.com/coqui-ai/STT/releases that makes me go "aha, that is the dotnet library nuget package".

Any pointers on getting this lib running via nuget package manager on Visual Studio on a .Net 6.0 application would be appreciated.

(That is assuming this is updated for the later versions of .Net. I know the original deepspeech repo appeared to be stalled on that front: https://github.com/mozilla/DeepSpeech/pull/3373 )

**Comment by reuben** [comment, marked as answer]
The .NET bindings code is still around in native_client/dotnet. But it's not being packaged and published in CI. If you're interested in helping getting it back up I can give some pointers on what needs to be done.

**Reply by rurounijones** [reply]
Unfortunately have too many things to work on already. Thanks for the information.

**Comment by patrelate** [comment]
Hi @reuben - by way of using @rurounijones project I have a vested interest in helping this along. I'm not a dev by trade (UAT manager by profession) but wondering if I could help move it along even if I'm googling+stack-overflow and having to learn some new things. Or if the complexity of issue is much deeper than what could done with that level of help.

**Reply by reuben** [reply]
The entire CI pipeline runs on GitHub Actions so you can do all the work independently by forking the repo and making changes to the CI definition in `.github/workflows/build-and-test.yml` and files referenced from there. I'm afraid with your level of expertise it'll take a very long time to get something up and running though. You'll need to write code/scripts to:

- Build the .NET bindings
- Package them and upload as artifacts
- Run them through integration tests

I'm happy to figure out NuGet.org publishing once that's done, but those steps involve knowledge of a lot of different tools and technologies. All of this process is defined in the old CI definition in the `taskcluster/` folder, but reading that requires knowledge of yet another tool. Let me know if there's anything I can help clarify!
